lib/Query.js

```diff
@@ -83,7 +83,11 @@ Query.prototype.exec = function (next) {
   };
 
   var indexName, index;
-  if(schema.hashKey.name !== this.query.hashKey.name) {
+  // Check both hash key and range key in the query to see if they do not match
+  // the hash and range key on the primary table.  If they don't match then we
+  // can look for a secondary index to query.
+  if(schema.hashKey.name !== this.query.hashKey.name ||
```

Comment:
So `.using` sets an option property, then I set `.IndexName` with what was specified in `.using` after all the logic to try to figure out what it should be.  This way the `.using` should always take precedence.

https://github.com/automategreen/dynamoose/pull/128/files#diff-c7f251522c2a751e0d457dee5ffb83f0R146